### ISSUE TITLE
Add toolchain setup check for OpenCL

### ISSUE THREAD

**Issue by andrjohns** [stan-dev/cmdstanr#858]
#### Summary

It was identified on the forums that OpenCL models can fail to compile without a helpful error message: https://discourse.mc-stan.org/t/cmdstan-installation-error/32900/5

I think we should add a configuration/setup step for OpenCL so we can catch the error and recommend running the setup function. I'm currently leaning towards adding it to `check_cmdstan_toolchain`, like:

```r
check_cmdstan_toolchain(fix = TRUE, opencl = TRUE)
```

Thoughts/prefs?

**Comment by jgabry** [comment]
Yeah I think adding it to `check_cmdstan_toolchain` is a good idea

**Comment by rok-cesnovar** [comment]
Agreed, great idea!
